File: kt/11.kt
class Solution {
    fun maxArea(height: IntArray): Int {
        val n = height.size
        var (left, right) = 0 to n - 1
        var ans = 0

        while (left < right) {
            val h = minOf(height[left], height[right])
            ans = maxOf(ans, h * (right - left))
            if (height[left] < height[right]) {
                left++
            } else {
                right--
            }
        }

        return ans
    }
}

File: py/11.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n = len(height)
        left, right = 0, n - 1
        ans = 0

        while left < right:
            ans = max(ans, (right - left) * min(height[left], height[right]))
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return ans
        